test/test_other.py

```diff
@@ -9365,12 +9365,12 @@ def test_standalone_system_headers(self):
         if cxx_only:
           create_file('a.cxx', inc)
```

Comment:
why does this test do the inclusion twice in 2 different files?

Reply:
I guess so that we know the headers don't define global symbols (that would then cause link failures when two objects contain the same symbol).    This change would obviously break that check, but I'm not sure how valuable it actually is.

Reply:
Ah right. In that case we can make the test even faster by including each file only once. Or just close this PR and keep the test as it is. I just noticed that the test was taking more than 300s and wondered why it was so slow.

Reply:
I think it might be reasonable to skip the double compilation + linking.

It looks like this was originally added in c3c9b6e462d0b973ce4f88ad0361e7d00a927f26, but it does seems like maybe overkill.